### PR TITLE
Js ms 02 exercises

### PR DESCRIPTION
**Pull request by JoeShep**
This commit begins the restructuring of Milestone 02. I have ordered the exercises for now, but anticipate slotting new ones in between some of these. Exercises 01-04 are ready for review. Others included here are just moved or renamed and will be updated later.

This also adds 4 challenge exercises. We can add more once we have a better grasp of how much we can realistically cover in this milestone. We can then push any overflow to the challenges section.

### CODE REVIEW

File: 02-db-driven-applications/exercises/02-express-static.md [hunk 1]
```diff
@@ -1,126 +0,0 @@
...
-│             │   \__/  /(_/     \__/ │             │
-│             │     (__/              │             │
-│             └───────────────────────┘             │
-│                                                   │
-│                     Sniffles                      │
-│                                                   │
-│                                                   │
-│                                                   │
-│                                                   │
-└───────────────────────────────────────────────────┘
```
Comment (C-Stein) [inline 02-db-driven-applications/exercises/02-express-static.md]: the loss of sniffles the cat ascii is very sad for me

File: 02-db-driven-applications/exercises/04-express-static.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+will receive the request, but pass it down the pipeline because it doesn't match
+the route. Next the `app.use(express.static(...))` handler will check if a file
+at the path `/public/test/index.html` exists. If so, it will send the file
+contents to the browser. If not, it will pass it further down the pipeline.
+However, at this point we do not have anything at this stage so the request will
+hang.
+
+## Requirements
+
+Create a Express server that uses static files. Place an `index.html`, an image
```
Comment (C-Stein) [inline 02-db-driven-applications/exercises/04-express-static.md]: this exercise comes before pug, right? So,we'll do basic api style app here, go to pug, and then come back to a more complex version of this?

File: 02-db-driven-applications/exercises/04-express-static.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+│             │   \__/  /(_/     \__/ │             │
+│             │     (__/              │             │
+│             └───────────────────────┘             │
+│                                                   │
+│                     Sniffles                      │
+│                                                   │
+│                                                   │
+│                                                   │
+│                                                   │
+└───────────────────────────────────────────────────┘
```
Comment (C-Stein) [inline 02-db-driven-applications/exercises/04-express-static.md]: Sniffles!

File: 02-db-driven-applications/exercises/05-express-middleware.md [hunk 4]
```diff
@@ -1,9 +1,7 @@
   # Express Middleware
 
 ## What is Middleware
-
-> Middleware is/are function(s) [that] run between the client request and the server answer. The most common middleware functionality needed are error managing, database interaction, getting info from static files or other resources.
-> ###### <em>[Sergio: From Stack Overflow](http://stackoverflow.com/users/2256325/sergio)</em>
+As we mentioned earlier, a middleware is a function that runs between the client request and the server answer. Let's dig a little deeper into middleware and
```
Comment (C-Stein) [inline 02-db-driven-applications/exercises/05-express-middleware.md]: and...?

**Comment by C-Stein** [comment]
overall looks very good. I made a few single line comments 👍

File: 02-db-driven-applications/exercises/04-express-static.md (hunk 2 above)
Comment (JoeShep) [inline 02-db-driven-applications/exercises/04-express-static.md]: Well, this is still serving up a static file (just not a file that gets js values injected into it). With the API we will be serving up JSON files.

File: 02-db-driven-applications/exercises/05-express-middleware.md (hunk 4 above)
Comment (JoeShep) [inline 02-db-driven-applications/exercises/05-express-middleware.md]: Yeah, this still needs to be completed. Next PR, for sure : )

**Pull request merged**